src/modules/sync/index.tsx

```diff
@@ -136,27 +135,53 @@ export function ContentTabSync( { selectedSite }: { selectedSite: SiteDetails }
 		localSiteId: selectedSite.id,
 		userId: user?.id,
 	} );
-	const {
-		syncSites,
-		isFetching: isFetchingSyncSites,
-		refetchSites,
-	} = useFetchWpComSites( connectedSites.map( ( { id } ) => id ) );
 	const [ connectSite ] = useConnectSiteMutation();
 	const [ disconnectSite ] = useDisconnectSiteMutation();
 	const { pushSite, pullSite, isAnySitePulling, isAnySitePushing } = useSyncSites();
+
+	const connectedSiteIds = connectedSites.map( ( { id } ) => id );
+	const {
+		data: syncSites = [],
+		isLoading: isLoadingSyncSites,
+		isFetching: isFetchingSyncSites,
+		isSuccess: isSuccessSyncSites,
+		isUninitialized: isUninitializedSyncSites,
+		refetch: refetchWpComSites,
+	} = useGetWpComSitesQuery( { connectedSiteIds, userId: user?.id } );
+
+	const refetchSites = useCallback( async (): Promise< SyncSite[] > => {
+		if ( isUninitializedSyncSites ) {
+			return [];
+		}
+		try {
+			const result = await refetchWpComSites();
+			return result.data ?? [];
+		} catch ( error ) {
+			// Query might not be ready to refetch yet (e.g., was skipped due to offline)
+			console.warn( 'Failed to refetch sites:', error );
+			return [];
+		}
```

Comment:
I see these pattern of catching the refetchWpComSite repeated three times. I wonder if makes sense to move the try catch inside the query and always return an empty array if it fails.

Reply:
Additionally, I have consolidated both buttons into one [with a condition](https://github.com/Automattic/studio/pull/2132/commits/7e1bbdbe6c10a200f90b7053b2613f0faa457f82#diff-9c7af3248cfe05390c0339be78082480629b738a1c274c90e4586c81b15c0128R40) that redirects to the Sync tab only when required in the commit [7e1bbdb](https://github.com/Automattic/studio/pull/2132/commits/7e1bbdbe6c10a200f90b7053b2613f0faa457f82)